src/sentry/migrations/0256_delete_codeowners_and_code_mappings_with_no_integration.py

```diff
@@ -0,0 +1,71 @@
+# Generated by Django 2.2.24 on 2021-12-07 10:56
+
+from django.db import migrations
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def delete_codeowners_with_no_integration(apps, schema_editor):
+    """
+    Delete the rows in the ProjectCodeOwners table that have a codemapping with null organization_integration_id.
+    """
+    ProjectCodeOwners = apps.get_model("sentry", "ProjectCodeOwners")
+
+    for code_owner in RangeQuerySetWrapperWithProgressBar(
```

Comment:
I think this only works with `ProjectCodeOwners.objects.all()`.

Reply:
I saw an example here: https://github.com/getsentry/sentry/blob/a38b45172b5f45f0d9fff518a11e4b36a84e536b/src/sentry/migrations/0245_delete_itunes_credentials.py#L23-L24

Reply:
^ yeah you wanna do the filtering within all

Reply:
Interesting 🤔 I've definitely tried to do this before and been told to do `.all()` instead

Reply:
honestly `RangeQuerySetWrapperWithProgressBar` might be overkill for this bc there is only 1 record in prod for this query.

Reply:
Conditions are ok if the fields have indexes. Without an index postgres will do a sequence scan on the rows which is roughly as slow as reading all the rows. It looks like `RepositoryProjectPathConfig.organization_integration_id` has an index.

Reply:
Just as a reminder: `RangeQuerySetWrapperWithProgressBar` iterates in chunks using `id > <last_id>`. So unless your indexes are written as `(id, <filter_field>)` it will essentially be using the id as an index, and then scanning for the rest of the fields.

If there is only 1 record then you could skip using `RangeQuerySetWrapperWithProgressBar` and just rely on any filters that are available.